index.bs

```diff
@@ -0,0 +1,225 @@
+</div>
+
+Evaluate layout instability {#sec-eval-layout-instability}
+--------------------------------------------------------
+
+<div algorithm="evaluate the layout instability">
+    When asked to <dfn export>evaluate the layout instability</dfn> given an active {{Document}} <var>doc</var>, run the following steps:
+
+    1. Let <var>elements</var> be the set of <a>shadow-including descendants</a> of the <var>doc</var>.
+    1. Let <var>unstableElements</var> be an empty list.
+    1. Let <var>totalWidth</var> be the current <a>visual viewport width</a>.
+    1. Let <var>totalHeight</var> be the current <a>visual viewport height</a>. 
+    1. Let <var>viewportDistance</var> be max(<var>totalWidth</var>, <var>totalHeight</var>).
+    1. For each <var>element</var> of <var>elements</var>:
+        1. Set <var>element</var>'s <a>current frame starting point</a> to the <a>starting point</a> of <var>element</var>.
+        1. Set <var>element</var>'s <a>current visual representation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>current frame starting point</a> and <a>previous frame starting point</a>. 
+        1. Set <var>element</var>'s <var>shiftFraction</var> to:
+            1. 0 if <a>previous frame starting point</a> is null.
+            1. The max distance the element has moved in any direction computed by:
+                1. Let <var>verticalDistance</var> be abs(<a>previous frame starting point</a>'s y-value - <a>current frame starting point</a>'s y-value) / <var>viewportDistance</var>.
+                1. Let <var>horizontalDistance</var> be abs(<a>previous frame starting point</a>'s x-value - <a>current frame starting point</a>'s x-value) / <var>viewportDistance</var>.
+                1. Take the max(<var>verticalDistance</var>, <var>horizontalDistance</var>).
+        1. If <var>unstable</var> is true, add a tuple to <var>unstableElements</var> with:
+            1. <a>current visual representation</a> of <var>element</var>
+            1. <a>previous visual representation</a> of <var>element</var>
+            1. <var>shiftFraction</var> of <var>element</var>:
+        1. Set <var>element</var>'s <a>previous frame starting point</a> point to <a>current frame starting point</a>.
+        1. Set <var>element</var>'s <a>previous visual representation</a> to the <a>current visual representation</a>.
+    1. Call <a>compute the layout shift</a> with <var>unstableElements</var>.
+</div>
+
+Compute the layout shift {#sec-compute-layout-shift}
+-----------------------------------------------------
+
+<div algorithm="compute the layout shift">
+    When asked to to <dfn export>compute the layout shift</dfn>, with <var>unstableElements</var>, a list of impacted <a>element</a>'s on the page, as input, run the following steps:
```

Comment:
This is not a list of impacted elements. It has a tuple which does not contain Element objects.

Reply:
ah good catch, updated.